### ISSUE TITLE
Hold queue orders until you are ready

### ISSUE THREAD

**Issue by Snurre86** [OpenRA/OpenRA#12408]
When you hold "shift" (default) to start a queue order the unit will start your first order immediately instead of waiting till you let go of the "shift". If you want the unit to start the queue order immediately you can always let go of the shift after one or two orders before reholding it again to queue more orders.

Here you see an example of the units waiting til you are ready for them to move and let go of the queue order button.
![Illustration of waypoints](http://i.imgur.com/oF5ICXE.jpg)

**Comment by penev92** [comment]
Duplicate of #2598 and partially #10470.

**Comment by tovl** [comment]
This does not sound like a duplicate to me. This is specifically asking to extend the order queueing functionality to put the queued orders on hold until the shift key is released again. That is different from the current behaviour, which immediately activates the first order regardless of if the shift key was held for the first order in the queue.

**Comment by abcdefg30** [comment]
The question now is if we wanted such a queuing feature ("don't start executing until exiting the queuing mode"), or if we consider what we currently have enough.

**Comment by tovl** [comment]
As long as that is an open question I think we should at least reopen this for discussion. Especially after all the recent improvements to queued orders this might be more relevant than it was when first posted.

**Comment by GraionDilach** [comment]
A positive aspect of this was it allowed better timing for cases like testing 1v1 unit performance.

**Comment by nr1995** [comment]
Yes the suggested behaviour is fairly vital functionality from vanilla.
If your unit starts moving the moment you enter the first waypoint,
then they could put themselves in danger long before you're finished defining their route.
Nothing should move until you let go of [Shift].

**Comment by pchote** [comment]
The original games (RA3 and C&C3, at least) made an explicit distinction between queued orders (which were issued immediately) and planning mode (which were issued at once when leaving the mode).

**Comment by nr1995** [comment]
> The original games (RA3 and C&C3, at least) made an explicit distinction between queued orders (which were issued immediately) and planning mode (which were issued at once when leaving the mode).

Will there be implemented a way to issue
“when-leaving-the-mode” commands like this in OpenRA?
